### PR TITLE
Repair W_AnalogRead after Data Refactor

### PR DESCRIPTION
**Pull request by retiutut**
Screen Shot 2020-05-02 at 8 54 24 PM

### CODE REVIEW

File: OpenBCI_GUI/W_AnalogRead.pde [hunk 1]
```diff
@@ -338,23 +339,41 @@ class AnalogReadBar{
...
+    void initArrays() {
+        nPoints = nPointsBasedOnDataSource();
+        timeBetweenPoints = (float)numSeconds / (float)nPoints;
+        analogReadPoints = new GPointsArray(nPoints);
+
+        for (int i = 0; i < nPoints; i++) {
+            float time = calcTimeAxis(i);
+            float analog_value = 0.0; //0.0 for all points to start
+            analogReadPoints.set(i, time, analog_value, "");
+        }
```
Comment (daniellasry) [inline OpenBCI_GUI/W_AnalogRead.pde:354]: Now that you have the initArray() function you need to delete lines 311 to 325 (inclusive) and replace them with a single call to initArrays().

Comment (daniellasry) [reply OpenBCI_GUI/W_AnalogRead.pde:354]: Also, delete the "arBuffSize" class variable since it's not used anymore.

Comment (daniellasry) [reply OpenBCI_GUI/W_AnalogRead.pde:354]: Also, delete the "analogReadData" calss variable.

Comment (daniellasry) [reply OpenBCI_GUI/W_AnalogRead.pde:354]: While you're there, please make all class variables "private". (or as many of them as possible right now)

File: OpenBCI_GUI/W_AnalogRead.pde [hunk 2]
```diff
@@ -338,23 +339,41 @@ class AnalogReadBar{
...
     void update() {
 
-        //update the voltage value text string
-        float val;
-
-        //update the voltage values
-        val = hub.validAccelValues[auxValuesPosition];
-        analogValue.string = String.format(getFmt(val),val);
+         // early out if unactive
+        if (auxValuesPosition == 1) {
```
Comment (daniellasry) [inline OpenBCI_GUI/W_AnalogRead.pde:362]: Why do you need this if statement? If auxValuesPosition is something other than 1 it should not early out?

File: OpenBCI_GUI/W_AnalogRead.pde [hunk 3]
```diff
@@ -369,48 +388,23 @@ class AnalogReadBar{
...
+        return -(float)numSeconds + (float)sampleIndex * timeBetweenPoints;
+    }
 
-            int voltage = dataPacketBuff[lastProcessedDataPacketInd].auxValues[auxValuesPosition];
+    void updatePlotPoints() {
+        List<double[]> allData = currentBoard.getData(nPoints);
+        int[] channels;
 
-            analogReadData[analogReadData.length - numSamplesToProcess + samplesProcessed] = voltage; //<>//
+        channels = analogBoard.getAnalogChannels();
```
Comment (daniellasry) [inline OpenBCI_GUI/W_AnalogRead.pde:399]: no need to declare and set channels onseparate lines

**Comment by retiutut** [comment]
AnalogRead Networking Data type will also need to be repaired.

**Comment by Andrey1994** [comment]
The only thing I dont like(and its almost everywhere in our code) is casting without type check....

**Comment by daniellasry** [comment]
> The only thing I dont like(and its almost everywhere in our code) is casting without type check....

Are you referring to `analogBoard = (AnalogCapableBoard) currentBoard;` ?
The widget is only created if the board is accelerometer capable, so we should never arrive here without the right type of board.

**Comment by daniellasry** [comment]
> > The only thing I dont like(and its almost everywhere in our code) is casting without type check....
> 
> Are you referring to `analogBoard = (AnalogCapableBoard) currentBoard;` ?
> The widget is only created if the board is accelerometer capable, so we should never arrive here without the right type of board.
> 
> Anywhere else, we should do a type check first.

Andrey and I talked offline and we agree this is fine because we do the type check in WidgetManager when instantiating the widget.

**Pull request merged**